modules/new_profile/jsx/NewProfileForm.js

```diff
@@ -0,0 +1,259 @@
+   * Additionaly add hiddenHeaders to global loris vairable
+   * for easy access by columnFormatter.
+   */
+    fetchData() {
+        $.ajax(
+            loris.BaseURL + '/new_profile/?format=json',
+            {
+                method: 'GET',
+                dataType: 'json',
+                success: (data) => {
+                    // FIXME: Remove the following line of code as soon as hiddenHeaders is
+                    // accepted as a prop by the StaticDataTable Component.
+                    this.setState(
+                        {
+                            configData: data,
+                            isLoaded: true,
+                        }
+                    );
+                },
+            }
+        );
+    }
+
+    componentWillMount() {
+        this.fetchData();
+    }
+
+    validate() {
+        const data = this.state.formData;
+        let isError = false;
+        if (data.dateTaken !== data.dateTakenConfirm) {
+            isError = true;
+        }
+        let decError = false;
+        if (this.state.configData['edc'] === 'true'
+            && data.edcDateTaken !== data.edcDateTakenConfirm
+        ) {
+            decError = true;
+        }
+        return isError || decError;
+    }
+
+
+    /**
+   * Handles form submission
+   *
+   * @param {event} e - Form submition event
+   */
+    handleSubmit(e) {
+        e.preventDefault();
+        const err = this.validate();
+        this.setState(
+                {
+                    errMessage: '',
+                }
+            );
+        if (!err) {
+            this.setState(
+                {
+                    isCreated: true,
+                }
+            );
+        } else {
+            this.setState(
+                {
+                    errMessage: 'Date of birth fields must match',
+                    isCreated: false,
+                }
+            );
+        }
+        $.ajax(
+            loris.BaseURL + '/new_profile/ajax/addProfile.php',
+            {
+                method: 'POST',
+                data: JSON.stringify(this.state.formData),
+                dataType: 'json',
+                success: (data) => {
+                    // FIXME: Remove the following line of code as soon as hiddenHeaders is
+                    // accepted as a prop by the StaticDataTable Component.
+                    data = data;
+                    this.setState(
+                        {
+                            newData: data,
+                        }
+                    );
+                },
+            }
+        );
+    }
+
+    /**
+   * Set the form data based on state values of child elements/componenets
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding form element
+   */
+    setFormData(formElement, value) {
+        let formData = this.state.formData;
+        formData[formElement] = value;
+
+        this.setState(
```

Comment:
this can be shortened to `this.setState({formData});`

Reply:
done